messaging/src/main/java/org/axonframework/queryhandling/DefaultQueryGateway.java

```diff
@@ -16,77 +16,63 @@
-import java.util.concurrent.CopyOnWriteArrayList;
 import java.util.concurrent.TimeUnit;
 import java.util.stream.Stream;
 
-import static java.util.Arrays.asList;
-import static org.axonframework.common.BuilderUtils.assertNonNull;
-
 /**
- * Implementation of the QueryGateway interface that allows the registration of dispatchInterceptors.
+ * Default implementation of the {@link QueryGateway} interface.
  *
- * @author Marc Gathier
  * @author Allard Buijze
+ * @author Marc Gathier
  * @author Steven van Beelen
- * @since 3.1
+ * @since 3.1.0
  */
 public class DefaultQueryGateway implements QueryGateway {
 
     private final QueryBus queryBus;
-    private final List<MessageDispatchInterceptor<? super QueryMessage>> dispatchInterceptors;
     private final MessageTypeResolver messageTypeResolver;
+    private final QueryPriorityCalculator priorityCalculator;
 
     /**
-     * Instantiate a {@link DefaultQueryGateway} based on the fields contained in the {@link Builder}.
-     * <p>
-     * Will assert that the {@link QueryBus} is not {@code null}, and will throw an {@link AxonConfigurationException}
-     * if it is {@code null}.
-     *
-     * @param builder the {@link Builder} used to instantiate a {@link DefaultQueryGateway} instance
-     */
-    protected DefaultQueryGateway(Builder builder) {
-        builder.validate();
-        this.queryBus = builder.queryBus;
-        this.dispatchInterceptors = builder.dispatchInterceptors;
-        this.messageTypeResolver = builder.messageTypeResolver;
-    }
-
-    /**
-     * Instantiate a Builder to be able to create a {@link DefaultQueryGateway}.
+     * Initialize the {@code DefaultQueryGateway} to send queries through the given {@code queryBus}.
      * <p>
-     * The {@code dispatchInterceptors} is defaulted to an empty list. The {@link QueryBus} is a
-     * <b>hard requirement</b> and as such should be provided.
+     * The {@link org.axonframework.messaging.QualifiedName names} for {@link QueryMessage QueryMessages} are resolved
+     * through the given {@code nameResolver}.
      *
-     * @return a Builder to be able to create a {@link DefaultQueryGateway}
+     * @param queryBus            The {@link QueryBus} to send queries on.
+     * @param messageTypeResolver The {@link MessageTypeResolver} resolving the
+     *                            {@link org.axonframework.messaging.QualifiedName names} for
+     *                            {@link QueryMessage QueryMessages} being dispatched on the {@code queryBus}.
+     * @param priorityCalculator  The {@link QueryPriorityCalculator} determining the priority of queries. Can be
+     *                            omitted.
      */
-    public static Builder builder() {
-        return new Builder();
+    public DefaultQueryGateway(@Nonnull QueryBus queryBus,
+                               @Nonnull MessageTypeResolver messageTypeResolver,
+                               @Nullable QueryPriorityCalculator priorityCalculator) {
```

Comment:
Question? Why we allow `Nullable` here? I see we have 
```    static QueryPriorityCalculator defaultQueryPriorityCalculator() {
        return query -> 0;
    }
```
So I prefer to use this instead of checking for null on the object. Is there any reason to leave it?